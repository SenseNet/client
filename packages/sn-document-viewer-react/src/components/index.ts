export * from './comment'
export * from './document-widgets'
export * from './page-widgets'
export * from './ConfirmationDialog'
export * from './DocumentViewer'
export * from './DocumentViewerError'
export * from './DocumentViewerLayout'
export * from './DocumentViewerLoading'
export * from './DocumentViewerRegeneratePreviews'
export * from './LayoutAppBar'
export * from './Page'
export * from './PageList'
export * from './styles'
