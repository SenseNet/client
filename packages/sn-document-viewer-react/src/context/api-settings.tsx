import React, { useEffect, useState } from 'react'
import { useRepository } from '@sensenet/hooks-react'
import { File as SnFile } from '@sensenet/default-content-types'
import { deepMerge, toNumber } from '@sensenet/client-utils'
import { v1 } from 'uuid'
import { Repository } from '@sensenet/client-core'
import {
  Annotation,
  CommentData,
  CommentWithoutCreatedByAndId,
  DocumentData,
  DocumentViewerApiSettings,
  Highlight,
  PreviewImageData,
  Redaction,
  Shape,
} from '../models'

/**
 * Adds a globally unique ID to the shape
 */
const addGuidToShape = <T extends Shape>(shape: T) => {
  shape.guid = v1()
  return shape
}

/**
 * Concatenates avatar url with host
 * @param {DocumentData} documentData
 */
function changeCreatedByUrlToCurrent(documentData: DocumentData): (value: CommentData) => CommentData {
  return (comment) => {
    return {
      ...comment,
      createdBy: { ...comment.createdBy, avatarUrl: `${documentData.hostName}${comment.createdBy.avatarUrl}` },
    }
  }
}

export const DocumentViewerApiSettingsContext = React.createContext<DocumentViewerApiSettings>({
  canEditDocument: async () => false,
  canHideRedaction: async () => false,
  canHideWatermark: async () => false,
  commentActions: {
    addPreviewComment: async ({ comment }) => ({ ...comment } as any),
    deletePreviewComment: async () => ({ modified: false }),
    getPreviewComments: async () => [],
  },
  getDocumentData: async () => ({
    documentName: 'example',
    documentType: '',
    fileSizekB: 3,
    hostName: '',
    idOrPath: 1,
    pageAttributes: [],
    pageCount: 0,
    shapes: { annotations: [], highlights: [], redactions: [] },
  }),
  getExistingPreviewImages: async () => [],
  isPreviewAvailable: async () => ({ Height: 0, Width: 0, Index: 0, Attributes: { degree: 0 } }),
  regeneratePreviews: async () => undefined,
  saveChanges: async () => undefined,
})

export const createDefaultApiSettings: (repo: Repository) => DocumentViewerApiSettings = (repo: Repository) => ({
  regeneratePreviews: async ({ document, abortController }) => {
    repo.executeAction({
      idOrPath: document.idOrPath,
      method: 'POST',
      body: undefined,
      name: 'RegeneratePreviews',
      requestInit: {
        signal: abortController.signal,
      },
    })
  },
  saveChanges: async ({ document, pages, abortController }) => {
    const reqBody = {
      Shapes: JSON.stringify([
        { redactions: document.shapes.redactions },
        { highlights: document.shapes.highlights },
        { annotations: document.shapes.annotations },
      ]),
      PageAttributes: JSON.stringify(
        pages
          .map((p) => (p.Attributes && p.Attributes.degree && { pageNum: p.Index, options: p.Attributes }) || undefined)
          .filter((p) => p !== undefined),
      ),
    }
    await repo.patch<SnFile>({
      idOrPath: document.idOrPath,
      content: reqBody,
      requestInit: {
        signal: abortController.signal,
      },
    })
  },
  getDocumentData: async (settings) => {
    const documentData = (
      await repo.load<SnFile>({
        idOrPath: settings.idOrPath,
        oDataOptions: {
          select: 'all',
        },
      })
    ).d

    return {
      idOrPath: settings.idOrPath,
      hostName: repo.configuration.repositoryUrl,
      fileSizekB: documentData.Size as number,
      pageCount: toNumber(documentData.PageCount, -1)!,
      documentName: documentData.DisplayName || '',
      documentType: documentData.Type || 'File',
      shapes: (documentData.Shapes && {
        redactions: (JSON.parse(documentData.Shapes)[0].redactions as Redaction[]).map((a) => addGuidToShape(a)) || [],
        annotations:
          (JSON.parse(documentData.Shapes)[2].annotations as Annotation[]).map((a) => addGuidToShape(a)) || [],
        highlights: (JSON.parse(documentData.Shapes)[1].highlights as Highlight[]).map((a) => addGuidToShape(a)) || [],
      }) || {
        redactions: [],
        annotations: [],
        highlights: [],
      },
      pageAttributes: (documentData.PageAttributes && JSON.parse(documentData.PageAttributes)) || [],
    }
  },
  isPreviewAvailable: async ({ document, version, page, abortController }) => {
    const responseBody = await repo.executeAction<{ page: number }, PreviewImageData & { PreviewAvailable: string }>({
      idOrPath: document.idOrPath,
      method: 'POST',
      name: `PreviewAvailable?version=${version}`,
      body: {
        page,
      },
      requestInit: {
        signal: abortController.signal,
      },
    })

    if (responseBody.PreviewAvailable) {
      responseBody.PreviewImageUrl = `${document.hostName}${responseBody.PreviewAvailable}`
      responseBody.ThumbnailImageUrl = `${document.hostName}${responseBody.PreviewAvailable.replace(
        'preview',
        'thumbnail',
      )}`
      return responseBody as PreviewImageData
    }
  },
  canEditDocument: async ({ document, abortController }) => {
    const response = await repo.security.hasPermission(document.idOrPath, ['Save'], undefined, {
      signal: abortController.signal,
      credentials: 'include',
    })
    return response
  },
  canHideRedaction: async ({ document, abortController }) =>
    await repo.security.hasPermission(document.idOrPath, ['PreviewWithoutRedaction'], undefined, {
      signal: abortController.signal,
      credentials: 'include',
    }),
  canHideWatermark: async ({ document, abortController }) =>
    await repo.security.hasPermission(document.idOrPath, ['PreviewWithoutWatermark'], undefined, {
      signal: abortController.signal,
      credentials: 'include',
    }),
  getExistingPreviewImages: async ({ document, version, abortController }) => {
    if (document.pageCount < -1) {
      throw Error('Preview generation error')
    }

    const response = await repo.executeAction({
      idOrPath: document.idOrPath,
      name: `GetExistingPreviewImages`,
      method: 'POST',
      body: {},
      oDataOptions: {
        select: 'all',
        expand: 'all',
        version,
      } as any,
      requestInit: {
        signal: abortController.signal,
      },
    })

    const availablePreviews = ((await response) as Array<PreviewImageData & { PreviewAvailable?: string }>).map((a) => {
      if (a.PreviewAvailable) {
        a.PreviewImageUrl = `${document.hostName}${a.PreviewAvailable}`
        a.ThumbnailImageUrl = `${document.hostName}${a.PreviewAvailable.replace('preview', 'thumbnail')}`
      }
      return a
    })

    const allPreviews: PreviewImageData[] = []
    for (let i = 0; i < document.pageCount; i++) {
      allPreviews[i] = availablePreviews[i] || ({ Index: i + 1 } as any)
      const pageAttributes = document.pageAttributes.find((p) => p.pageNum === allPreviews[i].Index)
      allPreviews[i].Attributes = pageAttributes && pageAttributes.options
    }
    return allPreviews
  },
  commentActions: {
    addPreviewComment: async ({ document, comment, abortController }) => {
      const response = await repo.executeAction<CommentWithoutCreatedByAndId, CommentData>({
        idOrPath: document.idOrPath,
        body: comment,
        name: 'AddPreviewComment',
        method: 'POST',
        requestInit: {
          signal: abortController.signal,
        },
      })
      return [response].map(changeCreatedByUrlToCurrent(document))[0]
    },
    deletePreviewComment: async ({ document, commentId, abortController }) => {
      const response = await repo.executeAction<any, { modified: boolean }>({
        idOrPath: document.idOrPath,
        body: { id: commentId },
        name: 'DeletePreviewComment',
        method: 'POST',
        requestInit: {
          signal: abortController.signal,
        },
      })
      return response
    },
    getPreviewComments: async ({ document, page, abortController }) => {
      const response = await repo.executeAction<any, CommentData[]>({
        idOrPath: document.idOrPath,
        name: 'GetPreviewComments',
        method: 'GET',
        oDataOptions: {
          page,
        } as any,
        requestInit: {
          signal: abortController.signal,
        },
      })
      return response.map(changeCreatedByUrlToCurrent(document))
    },
  },
})

export const DocumentViewerApiSettingsProvider: React.FC<{ options?: Partial<DocumentViewerApiSettings> }> = ({
  children,
  options,
}) => {
  const repo = useRepository()
  const [settings, setSettings] = useState(deepMerge(createDefaultApiSettings(repo), options))
  useEffect(() => {
    setSettings(deepMerge(createDefaultApiSettings(repo), options))
  }, [repo, options])

  return (
    <DocumentViewerApiSettingsContext.Provider value={settings}>{children}</DocumentViewerApiSettingsContext.Provider>
  )
}
