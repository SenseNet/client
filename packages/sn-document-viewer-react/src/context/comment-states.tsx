import React, { useState } from 'react'
import { DraftCommentMarker } from '@sensenet/client-core'

export interface CommentState {
  draft?: DraftCommentMarker
  setDraft: (d?: DraftCommentMarker) => void
  activeCommentId?: string
  setActiveComment: (commentId?: string) => void
}

export const defaultCommentState: CommentState = {
  setDraft: () => undefined,
  activeCommentId: undefined,
  setActiveComment: () => undefined,
}

export const CommentStateContext = React.createContext<CommentState>(defaultCommentState)

export const CommentStateProvider: React.FC = ({ children }) => {
  const [draft, setDraft] = useState<DraftCommentMarker>()
  const [activeCommentId, setActiveComment] = useState<string>()

  return (
    <CommentStateContext.Provider value={{ draft, setDraft, activeCommentId, setActiveComment }}>
      {children}
    </CommentStateContext.Provider>
  )
}
