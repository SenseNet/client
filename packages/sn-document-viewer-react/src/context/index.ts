export * from './api-settings'
export * from './comment-states'
export * from './comments'
export * from './document-data'
export * from './document-permissions'
export * from './localization-context'
export * from './preview-image-data'
export * from './viewer-settings'
export * from './viewer-state'
