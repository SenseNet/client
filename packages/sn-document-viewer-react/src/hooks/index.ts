export * from './use-comment-state'
export * from './use-comments'
export * from './use-document-data'
export * from './use-document-permissions'
export * from './use-document-viewer-api'
export * from './use-localization'
export * from './use-preview-image'
export * from './use-preview-images'
export * from './use-viewer-settings'
export * from './use-viewer-state'
