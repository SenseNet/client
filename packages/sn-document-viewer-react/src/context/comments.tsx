import React, { useEffect, useState } from 'react'
import { Trace } from '@sensenet/client-utils'
import { useDocumentData, useDocumentViewerApi, useViewerState } from '../hooks'
import { CommentData } from '../models/Comment'
import { DocumentViewerApiSettings } from '../models'

export type CommentsContextType = {
  comments: CommentData[]
} & DocumentViewerApiSettings['commentActions']

export const defaultCommentsContext: CommentsContextType = {
  comments: [],
  addPreviewComment: async () => undefined as any,
  deletePreviewComment: async () => undefined as any,
  getPreviewComments: async () => undefined as any,
}

export const CommentsContext = React.createContext<CommentsContextType>(defaultCommentsContext)

export interface CommentContextProviderProps {
  page: number
  images: 'preview' | 'thumbnail'
}

export const CommentsContextProvider: React.FC<CommentContextProviderProps> = ({ page, children, images }) => {
  const api = useDocumentViewerApi()
  const { documentData } = useDocumentData()
  const viewerState = useViewerState()
  const [comments, setComments] = useState<CommentData[]>([])
  useEffect(() => {
    const abortController = new AbortController()
    ;(async () => {
      if (viewerState.showComments && images === 'preview') {
        const result = await api.commentActions.getPreviewComments({ document: documentData, page, abortController })
        setComments(result)
      }
    })()
    return () => abortController.abort()
  }, [api.commentActions, documentData, images, page, viewerState.showComments])

  useEffect(() => {
    const disposables = [
      Trace.method({
        object: api.commentActions,
        isAsync: true,
        method: api.commentActions.addPreviewComment,
        onFinished: async ({ returned }) => {
          const a = await returned
          setComments([...comments, a])
          viewerState.updateState({ isPlacingCommentMarker: false })
        },
      }),
      Trace.method({
        object: api.commentActions,
        method: api.commentActions.deletePreviewComment,
        isAsync: true,
        onFinished: async ({ returned, methodArguments }) => {
          const returnedValue = await returned
          if (returnedValue.modified) {
            setComments(comments.filter(c => c.id !== methodArguments[0].commentId))
          }
        },
      }),
    ]

    return () => disposables.forEach(d => d.dispose())
  }, [api.commentActions, comments, viewerState])

  return (
    <CommentsContext.Provider
      value={{
        ...api.commentActions,
        comments,
      }}>
      {children}
    </CommentsContext.Provider>
  )
}
