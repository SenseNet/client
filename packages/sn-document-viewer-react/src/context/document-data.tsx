import { DocumentData } from '@sensenet/client-core'
import { deepMerge, DeepPartial, sleepAsync } from '@sensenet/client-utils'
import { useRepository } from '@sensenet/hooks-react'
import React, { useCallback, useEffect, useState } from 'react'
import Semaphore from 'semaphore-async-await'
import { POLLING_INTERVAL } from '../components'
import { PreviewState } from '../enums'
import { useDocumentViewerApi, useViewerSettings } from '../hooks'

const defaultDocumentData: DocumentData = {
  documentName: '',
  documentType: '',
  fileSizekB: 0,
  hostName: '',
  idOrPath: 0,
  pageCount: PreviewState.Loading,
  shapes: {
    annotations: [],
    highlights: [],
    redactions: [],
  },
  error: undefined,
}

export interface DocumentDataContextType {
  documentData: DocumentData
  updateDocumentData: (data: DeepPartial<DocumentData>) => void
  isInProgress: boolean
  triggerReload: () => void
}

export const defaultDocumentDataContextValue: DocumentDataContextType = {
  documentData: defaultDocumentData,
  updateDocumentData: () => undefined,
  isInProgress: false,
  triggerReload: () => {},
}

export const DocumentDataContext = React.createContext<DocumentDataContextType>(defaultDocumentDataContextValue)

export const DocumentDataProvider: React.FC = ({ children }) => {
  const api = useDocumentViewerApi()
  const doc = useViewerSettings()
  const repo = useRepository()

  const [documentData, setDocumentData] = useState<DocumentData>(defaultDocumentData)
  const [loadLock] = useState(new Semaphore(1))
  const [isInProgress, setIsInProgress] = useState(false)
  const [reloadToken, setReloadToken] = useState(0)

  const triggerReload = () => setReloadToken((prevValue) => prevValue + 1)

  useEffect(() => {
    const ac = new AbortController()
    const getData = async () => {
      const result = await api.getDocumentData({
        hostName: repo.configuration.repositoryUrl,
        idOrPath: doc.documentIdOrPath,
        version: doc.version,
        abortController: ac,
      })
      if (result.pageCount === PreviewState.Loading && ac.signal.aborted === false) {
        setDocumentData(result)
        await sleepAsync(POLLING_INTERVAL)
        getData()
      } else {
        setDocumentData(result)
      }
    }
    ;(async () => {
      try {
        setIsInProgress(true)
        await loadLock.acquire()
        await getData()
      } catch (error) {
        if (!ac.signal.aborted) {
          setDocumentData({ ...defaultDocumentData, pageCount: PreviewState.ClientFailure, error: error.toString() })
          throw error
        }
      } finally {
        setIsInProgress(false)
        loadLock.release()
      }
    })()
    return () => ac.abort()
  }, [api, doc.documentIdOrPath, doc.version, loadLock, repo.configuration.repositoryUrl, reloadToken])

  const updateDocumentData = useCallback(
    (newDocData: Partial<DocumentData>) => {
      const merged = deepMerge(documentData, newDocData)
      if (JSON.stringify(documentData) !== JSON.stringify(merged)) {
        setDocumentData(merged)
      }
    },
    [documentData],
  )

  if (!documentData.idOrPath) {
    return null
  }

  return (
    <DocumentDataContext.Provider value={{ documentData, updateDocumentData, isInProgress, triggerReload }}>
      {children}
    </DocumentDataContext.Provider>
  )
}
