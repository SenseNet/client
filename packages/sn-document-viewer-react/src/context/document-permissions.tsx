import React, { useEffect, useState } from 'react'
import { useDocumentData, useDocumentViewerApi } from '../hooks'

export const DocumentPermissionsContext = React.createContext({
  canEdit: false,
  canHideRedaction: false,
  canHideWatermark: false,
})

export const DocumentPermissionsContextProvider: React.FC = ({ children }) => {
  const { documentData } = useDocumentData()
  const api = useDocumentViewerApi()
  const [canEdit, setcanEdit] = useState(false)
  const [canHideRedaction, setcancanHideRedaction] = useState(false)
  const [canHideWatermark, setcanHideWatermark] = useState(false)

  useEffect(() => {
    const abortController = new AbortController()
    ;(async () => {
      try {
        const canEditPromise = api.canEditDocument({ document: documentData, abortController })
        const cancanHideRedactionPromise = api.canHideRedaction({ document: documentData, abortController })
        const canHideWatermarkPromise = api.canHideWatermark({ document: documentData, abortController })

        const [canEditValue, cancanHideRedactionValue, canHideWatermarkValue] = await Promise.all([
          canEditPromise,
          cancanHideRedactionPromise,
          canHideWatermarkPromise,
        ])
        setcanEdit(canEditValue)
        setcancanHideRedaction(cancanHideRedactionValue)
        setcanHideWatermark(canHideWatermarkValue)
      } catch (error) {
        if (!abortController.signal.aborted) {
          throw error
        }
      }
    })()
  }, [api, documentData])

  return (
    <DocumentPermissionsContext.Provider
      value={{
        canEdit,
        canHideRedaction,
        canHideWatermark,
      }}>
      {children}
    </DocumentPermissionsContext.Provider>
  )
}
