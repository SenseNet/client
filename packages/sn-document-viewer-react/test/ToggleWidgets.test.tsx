import IconButton from '@material-ui/core/IconButton/IconButton'
import { mount, shallow } from 'enzyme'
import React from 'react'
import {
  ToggleBase,
  ToggleCommentsWidget,
  ToggleRedactionWidget,
  ToggleShapesWidget,
  ToggleThumbnailsWidget,
  ToggleWatermarkWidget,
} from '../src/components/document-widgets'
import { defaultViewerState, ViewerStateContext } from '../src/context'
import { useViewerState } from '../src/hooks'

describe('Component', () => {
  it('ToggleBase should render without crashing', () => {
    const wrapper = shallow(
      <ToggleBase isVisible={true} title="Test" setValue={jest.fn()}>
        <span>Some children</span>
      </ToggleBase>,
    )
    expect(wrapper).toMatchSnapshot()
  })

  it('ToggleRedaction should render without crashing', () => {
    const wrapper = shallow(<ToggleRedactionWidget />)
    expect(wrapper).toMatchSnapshot()
  })

  it('ToggleShapes should render without crashing', () => {
    const wrapper = shallow(<ToggleShapesWidget />)
    expect(wrapper).toMatchSnapshot()
  })

  it('ToggleThumbnails should render without crashing', () => {
    const wrapperWithActiveStyle = mount(
      <ViewerStateContext.Provider value={{ ...defaultViewerState, showThumbnails: true }}>
        <ToggleThumbnailsWidget
          style={{
            fill: 'black',
          }}
          activeColor="pink"
        />
      </ViewerStateContext.Provider>,
    )

    expect(wrapperWithActiveStyle).toMatchSnapshot()
  })

  it('ToggleComments should render without crashing', () => {
    const wrapper = mount(
      <ViewerStateContext.Provider value={{ ...defaultViewerState, showComments: true }}>
        <ToggleCommentsWidget activeColor="pink" />
      </ViewerStateContext.Provider>,
    )

    expect(wrapper).toMatchSnapshot()
  })

  it('ToggleWatermark should render without crashing', () => {
    const wrapper = shallow(<ToggleWatermarkWidget />)
    expect(wrapper).toMatchSnapshot()
  })

  it('Click on toggle should call setValue with the opposite of isVisible', () => {
    const setValue = jest.fn()
    const wrapper = shallow(<ToggleBase isVisible={false} title="Test" setValue={setValue} />)
    wrapper.find(IconButton).simulate('click')
    expect(setValue).toBeCalledWith(true)
  })
})

it('Click on toggle should change the state of showThumbnails in viewer-state provider', () => {
  const updateState = jest.fn()

  const wrapper = mount(
    <ViewerStateContext.Provider
      value={{
        ...defaultViewerState,
        updateState,
      }}>
      <ToggleThumbnailsWidget />
    </ViewerStateContext.Provider>,
  )
  wrapper.find(IconButton).simulate('click')

  expect(updateState).toBeCalledWith({ showThumbnails: true })
})

it('Click on toggle should change the state of showComments in viewer-state provider', () => {
  const updateState = jest.fn()

  const wrapper = mount(
    <ViewerStateContext.Provider
      value={{
        ...defaultViewerState,
        updateState,
      }}>
      <ToggleCommentsWidget />
    </ViewerStateContext.Provider>,
  )
  wrapper.find(IconButton).simulate('click')

  expect(updateState).toBeCalledWith({ showComments: true })
})
