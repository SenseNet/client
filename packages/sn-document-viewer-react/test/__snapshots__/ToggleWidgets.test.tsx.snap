// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component ToggleBase should render without crashing 1`] = `
<div
  style={
    Object {
      "display": "inline-block",
    }
  }
>
  <WithStyles(ForwardRef(IconButton))
    color="primary"
    onClick={[Function]}
    style={
      Object {
        "opacity": 0.5,
      }
    }
    title="Test"
  >
    <span>
      Some children
    </span>
  </WithStyles(ForwardRef(IconButton))>
</div>
`;

exports[`Component ToggleRedaction should render without crashing 1`] = `
<Component
  isVisible={true}
  setValue={[Function]}
  title="Toggle redaction"
>
  <PictureInPictureIcon />
</Component>
`;

exports[`Component ToggleShapes should render without crashing 1`] = `
<Component
  isVisible={true}
  setValue={[Function]}
  title="Toggle shapes"
>
  <DashboardIcon />
</Component>
`;

exports[`Component ToggleThumbnails should render without crashing 1`] = `
<Component
  isVisible={false}
  setValue={[Function]}
  title="Toggle thumbnails"
>
  <svg
    enableBackground="new 0 0 1000 1000"
    fill="#ffffff"
    style={
      Object {
        "height": "1em",
        "width": "1em",
      }
    }
    version="1.1"
    viewBox="0 0 1000 1000"
    xmlns="http://www.w3.org/2000/svg"
  >
    <g>
      <path
        d="M826.7,990H173.3C83.5,990,10,916.5,10,826.7V173.3C10,83.5,83.5,10,173.3,10h653.3C916.5,10,990,83.5,990,173.3v653.3C990,916.5,916.5,990,826.7,990z M173.3,91.7c-44.9,0-81.7,36.8-81.7,81.7v653.3c0,44.9,36.7,81.7,81.7,81.7h653.3c44.9,0,81.7-36.8,81.7-81.7V173.3c0-44.9-36.8-81.7-81.7-81.7H173.3z"
      />
      <path
        d="M336.7,990c-24.5,0-40.8-16.3-40.8-40.8V50.8c0-24.5,16.3-40.8,40.8-40.8c24.5,0,40.8,16.3,40.8,40.8v898.3C377.5,973.7,361.2,990,336.7,990z"
      />
      <path
        d="M581.7,540.8c-12.2,0-20.4-4.1-28.6-12.3c-16.3-16.3-16.3-40.8,0-57.2l122.5-122.5c16.3-16.3,40.8-16.3,57.2,0s16.3,40.8,0,57.2L610.3,528.6C602.1,536.8,593.9,540.8,581.7,540.8z"
      />
      <path
        d="M704.2,663.3c-12.2,0-20.4-4.1-28.6-12.3L553.1,528.6c-16.3-16.3-16.3-40.8,0-57.2c16.3-16.3,40.8-16.3,57.2,0l122.5,122.5c16.3,16.3,16.3,40.8,0,57.2C724.6,659.3,716.4,663.3,704.2,663.3z"
      />
    </g>
  </svg>
</Component>
`;

exports[`Component ToggleWatermark should render without crashing 1`] = `
<Component
  isVisible={false}
  setValue={[Function]}
  title="Toggle watermark"
>
  <BrandingWatermarkIcon />
</Component>
`;
